### PR TITLE
bglibpy 4.7.2

### PR DESCRIPTION
**Pull request by anilbey**

### CODE REVIEW

**Review by arnaudon — DISMISSED** [review]

File: bluebrain/repo-bluebrain/packages/py-bglibpy/package.py [hunk 1]
```diff
@@ -13,17 +13,17 @@ class PyBglibpy(PythonPackage):
...
+    # dependencies from setup.py
     depends_on('py-numpy@1.8:', type='run')
     depends_on('py-matplotlib@3.0.3:', type='run')
     depends_on('py-cachetools', type='run')
-    depends_on('py-h5py@2.3:', type='run')
-
-    depends_on('py-bluepy@2.1:2.999', type='run')
-    depends_on('py-libsonata', type='run')
+    depends_on('py-bluepy@2.4.2:2.999', type='run')
+    depends_on('py-bluepy-configfile@0.1.18:', type='run')
```
Comment (matz-e) [inline bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: I think this clashes with `py-minis-validation`, which requires both `py-bglibpy` and `py-bluepy-configfile@0.1.10:0.999`.

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: But a pair of `py-bluepy` and `py-bluepy-configfile` versions that satisfy both `py-bglibpy` and `py-minis-validation` can be found I think.

`py-minis-validation` has
```
    depends_on('py-bluepy@2.0:2.999', type='run')
    depends_on('py-bluepy-configfile@0.1.10:0.999', type='run')
    depends_on('py-bglibpy@4.4:4.999', type='run')
```

`py-bglibpy` has

```
    depends_on('py-bluepy@2.4.2:2.999', type='run')
    depends_on('py-bluepy-configfile@0.1.18:', type='run')

```

e.g. using py-bluepy>=2.4.2,<2.999 and py-bluepy-configfile>=0.1.18,<0.999 will satisfy both packages.

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: I just added an upperbound to bluepy-configfile `depends_on('py-bluepy-configfile@0.1.18:0.999', type='run')` to try.

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: Adding the upperbound didn't fix it. On top of that it caused the `science` pipeline to fail.

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: I reverted that commit. The error in the NSE pipeline is this

```
Error: py-minis-validation%gcc ^neuron%intel is unsatisfiable, errors are:
  no version satisfies the given constraints
```

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: There's no `py-bluepy-configfile>=0.1.18 on `bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py` is that the cause? If it is so how come the rest of the pipelines succeed?

```
    version('develop', branch='main')
    version('0.1.17', tag='bluepy-configfile-v0.1.17')
    version('0.1.15', tag='bluepy-configfile-v0.1.15')
    version('0.1.14', tag='bluepy-configfile-v0.1.14')
    version('0.1.11', tag='bluepy-configfile-v0.1.11')
    version('0.1.7', tag='bluepy-configfile-v0.1.7')

```

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: Probably the upperbound on py-minis-validation is to exclude the `develop` version. The develop version is the only version that satisties >0.1.18

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: `py-bluepy-configfile>=0.1.18` is needed for the GluSynapse simulations.

@mgeplf @joni-herttuainen how should we do? Does it make sense to release a new `py-bluepy-configfile` or to release a `py-bluepy` that enforces `py-bluepy-configfile>=0.1.18`?

Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/py-bglibpy/package.py:26]: @anilbey, sorry just saw your comments now (tab stale 🤦 ). Yes, we'd need a new version of `py-bluepy-configfile`!

File: bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py [hunk 2]
```diff
@@ -13,6 +13,7 @@ class PyBluepyConfigfile(PythonPackage):
     git      = "git@bbpgitlab.epfl.ch:nse/bluepy-configfile.git"
 
     version('develop', branch='main')
+    version('0.1.18', tag='bluepy-configfile-v0.1.18')
```
Comment (anilbey) [inline bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py]: @matz-e @mgeplf do you know why this still doesn't work? The tag was created 2 days ago. 

* add0aee - (tag: bluepy-configfile-v0.1.18, origin/main, origin/HEAD) Version 0.1.18 (2 days ago) <Mike Gevaert>
* 2df43f7 - [BLPY-278]: add support for Conductance stimuli (2 days ago) <Michael Gevaert>

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py]: I updated py minis validation's bglibpy dependency to fix it. 

`depends_on('py-bglibpy@4.7.2:4.999', type='run')`

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py]: It again fails.
That's the error message. I do not see any conflict between those constraints.

```
  imposed_constraint(4468,"version_satisfies","py-bluepy-configfile","0.1.18:")
  imposed_constraint(5217,"version_satisfies","py-bluepy-configfile","0.1.10:0.999")
```

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py]: Does `"py-bluepy-configfile","0.1.18:"` include or exclude version `0.1.18`?

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py]: Can one not add 2 updates (one for bglibpy one for bluepy-configfile) within one change?

Comment (anilbey) [reply bluebrain/repo-bluebrain/packages/py-bluepy-configfile/package.py]: Do I need to add bluepy-configfile in another PR and let it be merged to the develop prior to this PR?

**Comment by anilbey** [comment]
Tests pass after the rebase.

File: bluebrain/deployment/gitlab-ci.yml [hunk 3]
```diff
@@ -225,7 +225,7 @@ github_information:
     # Activate if checksums fail
     # - spack clean -a
     # Concretize and pre-fetch, afterwards build the environment
-    - spack concretize -f|tee "concretization_${DEPLOY_ENV}.txt"
+    - spack --show-cores=full concretize -f|tee "concretization_${DEPLOY_ENV}.txt"
```
Comment (mgeplf) [inline bluebrain/deployment/gitlab-ci.yml:228]: Everything looks good to me, but I think @matz-e should look at this.

Comment (matz-e) [reply bluebrain/deployment/gitlab-ci.yml:228]: That was me :)

"This is fine"

**Review by matz-e — APPROVED** [review]

File: bluebrain/deployment/gitlab-ci.yml (hunk 3 above)
Comment (anilbey) [reply bluebrain/deployment/gitlab-ci.yml:228]: Alright, merging. Thank you both!

**Pull request merged**